homeassistant/components/voip/assist_satellite.py

```diff
@@ -274,53 +275,77 @@ async def _do_announce(
             rtp_port=self._rtp_port,
         )
 
-        # Check if caller hung up or didn't pick up
-        self._check_announcement_ended_task = (
+        # Check if caller didn't pick up
+        self._check_announcement_pickup_task = (
             self.config_entry.async_create_background_task(
                 self.hass,
-                self._check_announcement_ended(),
-                "voip_announcement_ended",
+                self._check_announcement_pickup(),
+                "voip_announcement_pickup",
             )
         )
 
         try:
-            await self._announcement_future
+            await self._call_end_future
         except TimeoutError:
             # Stop ringing
+            _LOGGER.debug("Caller did not pick up in time")
             sip_protocol.cancel_call(call_info)
             raise
 
-    async def _check_announcement_ended(self) -> None:
+    async def _check_announcement_pickup(self) -> None:
         """Continuously checks if an audio chunk was received within a time limit.
 
-        If not, the caller is presumed to have hung up and the announcement is ended.
+        If not, the caller is presumed to have not picked up the phone and the announcement is ended.
         """
-        while self._announcement is not None:
+        while True:
             current_time = time.monotonic()
             if (self._last_chunk_time is None) and (
                 (current_time - self._announcment_start_time)
                 > _ANNOUNCEMENT_RING_TIMEOUT
             ):
                 # Ring timeout
+                _LOGGER.debug("Ring timeout")
                 self._announcement = None
-                self._check_announcement_ended_task = None
-                self._announcement_future.set_exception(
+                self._check_announcement_pickup_task = None
```

Comment:
Should we also cancel this task?

Reply:
we always should make sure that any task we spawn, stops or is cancelled. You can check if a task is done by calling `.done()` on it.

Reply:
In this case that task is what is currently executing, and will soon end at this point. It seems odd that a task would cancel itself instead of just let itself terminate normally?